pkg/web/http_receiver_v2_test.go

```diff
@@ -24,157 +24,177 @@ import (
 func TestForwardingEndToEndV2(t *testing.T) {
 	t.Parallel()
 
-	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
```

Comment:
This diff is particularly difficult to read. This large red block has simply been nested inside the `t.Run` call (see larg green block below).